modules/aioseop_sitemap.php

```diff
@@ -1468,7 +1472,22 @@ public function sitemap_output_hook( $query ) {
 				if ( $gzipped ) {
 					ob_start();
 				}
-				$this->do_rewrite_sitemap( $sitemap_type, $page );
+
+				$content = $this->do_rewrite_sitemap( $sitemap_type, $page );
+
+				// if the sitemap has no content, it's probabaly invalid and is being called directly.
+				// @issue https://github.com/semperfiwebdesign/all-in-one-seo-pack/issues/2190
+				if ( empty( $content ) ) {
```

Comment:
Was there any other way to make certain the post exists?

I'm not sure if you have a post id available to use with [post_exists()](https://developer.wordpress.org/reference/functions/post_exists/).